src/backend/api/meals.js

```diff
@@ -2,13 +2,119 @@ const express = require("express");
+    };
+
   try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
+
+    // api/meals?maxPrice=90
+    if (maxPrice) {
+      const result = await knex
+        .select("*")
+        .from("meal")
+        .where("price", "<", +maxPrice);
+
+      response.data = result;
+    }
+
+    //api/meals?availableReservations=true
+    if (availableReservations===true) {
+      const result = await knex
+      .select('meal_id', 'meal_title', 'max_reservations', knex
+      .raw('-SUM(number_of_guests)'), knex
+      .raw('max_reservations - SUM(number_of_guests)'))
+      .from('meal')
+      .leftJoin('reservation', 'meal_id', 'reservation.meal_id')
+      .groupBy('meal_id')
+      .having(knex.raw('SUM(number_of_guests) < max_reservations'));
+
+      response.data = result;
+    } else {
+      result = await knex
+      .select('meal_id', 'meal_title', 'max_reservations', knex
+      .raw('-SUM(number_of_guests)'), knex
+      .raw('max_reservations - SUM(number_of_guests)'))
+      .from('meal')
+      .leftJoin('reservation', 'meal_id', 'reservation.meal_id')
+      .groupBy('meal_id')
+      .having(knex.raw('SUM(number_of_guests) > max_reservations'));
+
+      response.data = result;
+    }
+
+    //api/meals?title=Rød%20grød
+    if (title) {
+      const result = await knex("meals")
+      .select("*")
+      .where("title", "like", `%${title}%`);
+
+      response.data = result;
+    }
+
+    //api/meals?dateAfter=2022-10-01
+    if (dateAfter) {
+      const result = await knex
+        .select("*")
+        .from("meals")
+        .where("meal_time", ">", new Date(dateAfter));
+
+      response.data = result;
+    }
+
+    //api/meals?dateBefore=2022-08-08
+    if (dateBefore) {
+      const result = await knex("meals")
+      .select("*")
+      .where("meal_time", "<", dateBefore);
+
+      response.data = result;
+    }
+
+    //api/meals?limit=7
+    if (limit) {
+      const result = await knex("meal")
+        .select("*")
+        .limit(+limit);
+
+      response.data = result;
+    }
+
+    //api/meals?sortKey=price
+    if (["meal_time", "max_reservations", "price"].includes(sortKey)) {
+      const result = await knex("meal")
+        .select("*")
+        .orderBy(sortKey, sortDir === "desc" ? sortDir : "asc");
+
+      response.data = result;
+    }
+
+    //api/meals?sortKey=price&sortDir=desc
+    if (["meal_time", "max_reservations", "price"].includes(sortKey)) {
+
+      if (sortDir === "asc" || sortDir === "desc") {
+        const result = await knex("meal")
+          .select("*")
+          .orderBy(sortKey, sortDir);
+
+        response.data = result;
+      }
+    }
+
+    res.status(response.status).json({ data: response.data, message: response.message });
   } catch (error) {
-    throw error;
+    console.error("Server Error", error);
+    res.status(500).json({ error: "ERROR" });
+ 
```

Comment:
If I am not mistaken you are missing the method that would handle the `/api/meals/:id/reviews` endpoint, which was requested by the assignment. This might have been just due to you overlooking this requirement, as it is quite easy to implement. 😄 Bear in mind though that even if you have to read the `reviews` table, the method should be here in the `meals.js` file, beacuse the path starts with `api/meals`.